source/loader/layers/validation/backtrace_lin.cpp

```diff
@@ -0,0 +1,39 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+#include "backtrace.hpp"
+
+#include <execinfo.h>
+#include <vector>
+
+namespace validation_layer {
+
+std::vector<BacktraceLine> getCurrentBacktrace() {
+    void *backtraceFrames[MAX_BACKTRACE_FRAMES];
+    int frameCount = backtrace(backtraceFrames, MAX_BACKTRACE_FRAMES);
+    char **backtraceStr = backtrace_symbols(backtraceFrames, frameCount);
+
+    if (backtraceStr == nullptr) {
+        return std::vector<BacktraceLine>(1, "Failed to acquire a backtrace");
+    }
+
+    std::vector<BacktraceLine> backtrace;
+    try {
+        for (int i = 0; i < frameCount; i++) {
+            backtrace.emplace_back(backtraceStr[i]);
```

Comment:
nit: You can already `reserve()` memory for 64 entries in the vector beforehand, avoiding vector copying when increasing its size.

Reply:
I wouldn't worry about things like that in non-performance-sensitive code. It just muddies things.